### ISSUE TITLE
Reflective access warning re: Parameter.index

### ISSUE THREAD

**Issue by igorcmoraes** [pholser/junit-quickcheck#260]
Maybe this problem is caused by a mistake in configuration, because I'm not familiar with PBT (not even with java, actually). So I will describe all the process, from the creation of the files to the execution.

I created a maven project, and added the junit-quickcheck dependence of mvnrepository in pom.xml. Then I created a class in test folder with the basic example (the example in README.md of this project) and run as JUnit Test.

So appears an error about SLF4J, and I fixed it by adding this dependencies: 

```
<dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-api</artifactId>
    <version>1.7.5</version>
</dependency>
<dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-log4j12</artifactId>
    <version>1.7.5</version>
</dependency>
```

Running after that, appears this message:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.pholser.junit.quickcheck.internal.Reflection (file:/home/huebr/.m2/repository/com/pholser/junit-quickcheck-core/0.9.1/junit-quickcheck-core-0.9.1.jar) to field java.lang.reflect.Parameter.index
WARNING: Please consider reporting this to the maintainers of com.pholser.junit.quickcheck.internal.Reflection
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

There is, guys... I made a mistake? I missed something? Thanks in advance!

**Comment by pholser** [comment]
This is likely due to my introducing a change that won't fly on newer JDKs. I have been developing and running using JDK 8 only...it seems that with newer versions, certain reflection operations will be discouraged or disallowed.

I'll see about addressing this soon.

**Comment by pholser** [comment]
This is likely due to my introducing a change that won't fly on newer JDKs. I have been developing and running using JDK 8 only...it seems that with newer versions, certain reflection operations will be discouraged or disallowed.

I'll see about addressing this soon.

**Comment by pholser** [comment]
@igorcmoraes So can I can faithfully reproduce your circumstances -- what version of the JDK are you running your examples on? What version of junit-quickcheck? Thanks!

**Comment by pholser** [comment]
@igorcmoraes never mind -- I'm able to see the warning appear with junit-quickcheck 0.9.1, JDK 13.0.1, and a test project I use (https://github.com/pholser/junit-quickcheck-scratchpad). I'll update both to correct. It's not necessary to use reflection to get the index of the parameter I want.

**Comment by pholser** [comment]
@igorcmoraes This should be corrected in version 0.9.2. Thanks for reporting!
